### PR TITLE
Implement MediaPlayer.Dispose()

### PR DESCRIPTION
**Pull request by fcantin-nventive**
## PR Type
What kind of change does this PR introduce?
Feature implementation

## What is the current behavior?
MediaPlayer.Dispose() is wrongly marked as NotImplemented, which generates a warning in debug/an error in release at build time.

## What is the new behavior?
MediaPlayer.Dispose() can be called, in order to stop the video when you leave its page.

## Other information
Internal work item: 135799

### CODE REVIEW

**Comment by davidjohnoliver** [comment]
"in order to stop the video when you leave its page" Does it... do that? Given that the override you added for iOS literally does nothing, and the one on Android seems to do something else?

**Comment by fcantin-nventive** [comment]
> "in order to stop the video when you leave its page" Does it... do that? Given that the override you added for iOS literally does nothing, and the one on Android seems to do something else?

It does, yes. For one thing, I've updated the code of the PR to add cleanup code on the iOS side. Furthermore, it is important to call Dispose so that the video stops on both Android and UWP, otherwise the video's audio track keeps playing.

File: src/Uno.UWP/Media/Playback/MediaPlayer.iOS.cs [hunk 1]
```diff
@@ -379,6 +379,12 @@ private void OnSeekCompleted(bool finished)
 		internal void UpdateVideoGravity(AVLayerVideoGravity gravity)
 		{
 			_videoLayer.VideoGravity = gravity;
+			this.Dispose();
```
Comment (davidjohnoliver) [inline src/Uno.UWP/Media/Playback/MediaPlayer.iOS.cs]: Why do you call Dispose() here?

Comment (fcantin-nventive) [reply src/Uno.UWP/Media/Playback/MediaPlayer.iOS.cs]: There's no good reason, it must be a leftover from tests. I'll remove it.

**Pull request merged**